tests/kafkatest/tests/log4j_appender_test.py

```diff
@@ -25,27 +28,50 @@
 TOPIC = "topic-log4j-appender"
 MAX_MESSAGES = 100
 
-class Log4jAppenderTest(KafkaTest):
+class Log4jAppenderTest(Test):
     """
     Tests KafkaLog4jAppender using VerifiableKafkaLog4jAppender that appends increasing ints to a Kafka topic
     """
     def __init__(self, test_context):
-        super(Log4jAppenderTest, self).__init__(test_context, num_zk=1, num_brokers=1, topics={
+        super(Log4jAppenderTest, self).__init__(test_context)
+        self.num_zk = 1
+        self.num_brokers = 1
+        self.topics = {
             TOPIC: {'partitions': 1, 'replication-factor': 1}
-        })
-        self.num_nodes = 1
+        }
 
-        self.appender = KafkaLog4jAppender(self.test_context, self.num_nodes, self.kafka, TOPIC, MAX_MESSAGES)
-        self.consumer = ConsoleConsumer(self.test_context, num_nodes=self.num_nodes, kafka=self.kafka, topic=TOPIC, consumer_timeout_ms=1000)
+        self.zk = ZookeeperService(test_context, self.num_zk)
 
-    def test_log4j_appender(self):
+    def setUp(self):
+        self.zk.start()
+
+    def start_kafka(self, security_protocol, interbroker_security_protocol):
+        self.kafka = KafkaService(
+            self.test_context, self.num_brokers,
+            self.zk, security_protocol=security_protocol,
+            interbroker_security_protocol=interbroker_security_protocol, topics=self.topics)
+        self.kafka.start()
+
+    def start_appender(self, security_protocol):
+        self.appender = KafkaLog4jAppender(self.test_context, self.num_brokers, self.kafka, TOPIC, MAX_MESSAGES,
+                                           security_protocol=security_protocol)
+        self.appender.start()
+
+    def start_consumer(self, security_protocol):
+        self.consumer = ConsoleConsumer(self.test_context, num_nodes=self.num_brokers, kafka=self.kafka, topic=TOPIC,
+                                        consumer_timeout_ms=1000, security_protocol=security_protocol)
+
+    @matrix(security_protocol=['PLAINTEXT', 'SSL'])
+    def test_log4j_appender(self, security_protocol='PLAINTEXT'):
         """
         Tests if KafkaLog4jAppender is producing to Kafka topic
         :return: None
         """
-        self.appender.start()
+        self.start_kafka(security_protocol, security_protocol)
+        self.start_appender(security_protocol)
         self.appender.wait()
 
+        self.start_consumer(security_protocol)
         t0 = time.time()
```

Comment:
Probably not necessary to report consumer startup time in this test? But if you do, `t0 = time.time()` should go above `start_consumer`

Also - on line 88, change `self.consumer.stop_node(node)` -> `self.consumer.stop()`